src/ServiceControl/Recoverability/Grouping/Groupers/ClassifiableMessageDetails.cs

```diff
@@ -3,24 +3,33 @@ namespace ServiceControl.Recoverability
     using System.Linq;
     using ServiceControl.Contracts.Operations;
     using ServiceControl.MessageFailures;
+    using static ServiceControl.MessageFailures.FailedMessage;
 
     public struct ClassifiableMessageDetails
     {
+        public ProcessingAttempt ProcessingAttempt { get; }
         public FailureDetails Details { get; }
         public string MessageType { get; }
 
         public ClassifiableMessageDetails(FailedMessage message)
         {
-            var last = message.ProcessingAttempts.Last();
+            ProcessingAttempt = message.ProcessingAttempts.Last();
 
-            Details = last.FailureDetails;
-            MessageType = (string)last.MessageMetadata["MessageType"];
+            Details = ProcessingAttempt.FailureDetails;
+            MessageType = (string)ProcessingAttempt.MessageMetadata["MessageType"];
         }
 
         public ClassifiableMessageDetails(string messageType, FailureDetails failureDetails)
```

Comment:
Can we get rid of this constructor? It's only used for tests and for the data migration. What if the user is updating from a version that requires the data migration? They will not receive the new classifications currently.